cloud/satellite/satellite_activation_keys.py

```diff
@@ -0,0 +1,365 @@
+        description:
+            - The maximum number of registered hosts.
+        default: null
+        required: false
+    unlimited_content_hosts:
+        description:
+            - Describe if the activation key can have unlimited hosts.
+        default: null
+        required: false
+    release_version:
+        description:
+            - Content release version of the authorization key.
+        default: null
+        required: false
+    service_level:
+        description:
+            - Service level of the authorization key.
+        default: null
+        required: false
+    auto_attach:
+        description:
+            - Auto attach subscription upon registration.
+        default: null
+        required: false
+'''
+
+EXAMPLES = '''
+Example playbook entries using the satellite_activation_keys module to manage Satellite activation keys.
+
+    tasks:
+
+    - name: create new key
+      satellite_activation_keys:
+        user: admin
+        pwd: secret
+        satellite: "satellite.example.com"
+        key_action: create
+        key_name: mynewkey
+        organization: my_organization
+'''
+
+RETURN = '''
+msg:
+    description: status response of the REST call
+    returned: success
+    type: string
+    sample: {\"organization\":{\"name\":\"myorg\"}}
+'''
+
+def get_content_view_id(module,orga_id):
+    """Looks up content view ID for a given content view name.
+    """
+
+    url_action = 'https://' + module.params.get('satellite') + '/katello/api/v2/content_views/'
+    payload = {"organization_id":orga_id,"name":module.params.get('content_view')}
+    headers = {'Content-Type':'application/json'}
+
+    resp = open_url(url_action,method="GET",headers=headers,url_username=module.params.get('user'),url_password=module.params.get('pwd'),force_basic_auth=True,data=json.dumps(payload),validate_certs=module.params.get('validate_certs'))
+    resp_json = json.loads(resp.read())
+
+    if resp_json["total"] == 0:
+        module.fail_json(msg="Content View %s not found." % module.params.get('content_view'))
+    elif resp_json["total"] > 1:
+        module.fail_json(msg="Content View name %s not specific enough, more than one found." % module.params.get('content_view'))
+    cv_id = resp_json["results"][0]["id"]
+
+    return cv_id
+
+def get_environment_id(module,orga_id):
+    """Looks up environment ID for a given environment name.
+    """
+
+    env_name = module.params.get('environment')
+
+    url_action = 'https://' + module.params.get('satellite') + '/katello/api/v2/environments/'
+    payload = {"organization_id":orga_id,"name":env_name}
+    headers = {'Content-Type':'application/json'}
+
+    resp = open_url(url_action,method="GET",headers=headers,url_username=module.params.get('user'),url_password=module.params.get('pwd'),force_basic_auth=True,data=json.dumps(payload),validate_certs=module.params.get('validate_certs'))
+    resp_json = json.loads(resp.read())
+
+    if resp_json["total"] == 0:
```

Comment:
The problem is that there are small, but important differences between error handling in the functions. It would require at least two small helper functions, but I will look into it.